packages/terra-tabs/src/common-tabs/CommonTabs.jsx

```diff
@@ -0,0 +1,137 @@
+   * Returns the associated itemKey and metaData. e.g. onRequestActivate(itemKey, metaData)
+   */
+  onRequestActivate: PropTypes.func.isRequired,
+  /**
+   *  @private
+   * The style to be applied to the tabs
+   */
+  variant: PropTypes.oneOf(['workspace', 'framework']),
+  /**
+   * Indicates if tabs should fill the width available in the tab bar.
+   */
+  tabFill: PropTypes.bool,
+  /**
+   * Callback function when selection has changed.
+   * Parameters: 1. Event 2. Selected pane's key
+   */
+  onChange: PropTypes.func,
+};
+
+const getTabId = (id, itemKey) => `${id}-${itemKey}`;
+
+const getAssociatedPanelId = (id, itemKey) => `${getTabId(id, itemKey)}-panel`;
+
+const CommonTabs = ({
+  id,
+  activeItemKey,
+  activeSize,
+  children,
+  onRequestActivate,
+  variant,
+  onChange,
+  ...customProps
+}) => {
+  const theme = React.useContext(ThemeContext);
+
+  const [commonTabsContainerRef, commonTabsPortalsRef] = usePortalManager(activeItemKey);
+
+  const ariaLabel = 'workspace';
+
+  const tabData = React.Children.map(children, child => ({
+    id: getTabId(id, child.props.itemKey),
+    itemKey: child.props.itemKey,
+    associatedPanelId: getAssociatedPanelId(id, child.props.itemKey),
+    label: child.props.label,
+    icon: child.props.icon,
+    isIconOnly: child.props.isIconOnly,
+    customDisplay: child.props.customDisplay,
+    isSelected: child.props.itemKey === activeItemKey,
+    onSelect: onRequestActivate,
+    metaData: child.props.metaData,
+    isDisabled: child.props.isDisabled,
+  }));
+
+  const tabsClassNames = classNames(cy(
+    // 'tabs-container',
+    // { 'tab-fill': tabFill },
+    'structural',
+    theme.className,
+  ),
+  customProps.className);
+
+  return (
+    <div
+      className={variant === 'framework' ? tabsClassNames : cx('workspace')}
+      role="none"
+    >
+      <div aria-hidden className={cx('body-shadow-container')}>
+        <div className={cx('body-shadow')} />
+      </div>
+      <div role="none" className={cx('tab-header')}>
+        <Tabs variant={variant} ariaLabel={ariaLabel} tabData={tabData} onChange={onChange} />
+      </div>
+      <div role="none" className={cx('body')} ref={commonTabsContainerRef}>
+        {React.Children.map(children, child => {
+          let portalElement = commonTabsPortalsRef.current[child.props.itemKey]?.element;
+          if (!portalElement) {
+            portalElement = getPortalElement();
+            portalElement.setAttribute('role', 'none');
+
+            commonTabsPortalsRef.current[child.props.itemKey] = {
+              element: portalElement,
+            };
+          }
+
+          return (
+            React.cloneElement(child, {
+              key: child.props.itemKey,
+              id: getTabId(id, child.props.itemKey),
+              associatedPanelId: getAssociatedPanelId(id, child.props.itemKey),
+              isActive: child.props.itemKey === activeItemKey,
+              portalElement,
```

Comment:
I think we need to exclude portalElement for `framework` tabs variant.

Reply:
Conditionally passing portalElement as per variant: https://github.com/cerner/terra-framework/pull/1678/commits/b9e86635c0835a19ddf3d74c32c58023eaa703c0